app/Controllers/UserLogin.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Controllers;
+
+use CodeIgniter\API\ResponseTrait;
+use App\Models\UserModel;
+
+class UserLogin extends BaseController
+{
+    use ResponseTrait;
+
+    public function index()
+    {
+        if(session()->get('userData')) {
+            return redirect()->to("/home");
+        } else {
+            return view('pages/visitor_home');
+        }
+    }
+
+    public function home()
+    {
+        return view('pages/user_home', session()->userData);
+    }
+
+    public function login()
+    {
+        $request = \Config\Services::request();
+        $data = $request->getPost();
+
+        $email = $data['email'];
+        $password = $data['password'];
+        if($email == null || $password == null) {
+            return $this->response->setStatusCode(400)->setJSON("需帳號密碼進行登入");
```

Comment:
You already used the responseTrait, you can use the $this->respond() function.
And the $this->respond() is already implemented the setting the status code work.